src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private Figure[] figures = {
+            new Circle(0),
+            new IsoscelesTrapezoid(0, 0, 0),
+            new Rectangle(0, 0),
+            new Square(0),
+            new RightTriangle(0, 0)};
+
+    public Figure getRandomFigure() {
+        Figure figure = figures[random.nextInt(figures.length)];
+        return figure.createRandomFigure();
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. Also, ensure that you are not generating random values inside figure classes, but rather in the `FigureSupplier`.